### PR TITLE
Add flash attention as custom op so dynamo can trace it

### PR DESCRIPTION
**Pull request by JackCaoG**

### CODE REVIEW

File: torch_xla/experimental/custom_kernel.py [hunk 1]
```diff
@@ -183,3 +185,34 @@ def flash_attention(
...
+                        v: torch.Tensor,
+                        causal: bool = False):
+  return flash_attention(q, k, v, causal=causal)
+
+
+@impl(XLA_LIB, "flash_attention", "CompositeExplicitAutograd")
+def flash_attention_non_xla(q: torch.Tensor,
+                            k: torch.Tensor,
+                            v: torch.Tensor,
+                            causal: bool = False):
```
Comment (JackCaoG) [inline torch_xla/experimental/custom_kernel.py:207]: I didn't add `segment_ids ` as it is not implemented in `flash_attention`. Maybe I should add it here... @alanwaketan is `flash_attention` also a `torch.tensor`?

Comment (alanwaketan) [reply torch_xla/experimental/custom_kernel.py:207]: I don't think we need to support segment_ids. We can safely ignore it. I will drop the comment and the parameter later.

**Comment by JackCaoG** [comment]
@alanwaketan can I get a review for this one?

**Review by alanwaketan — APPROVED** [review]
LGTM.

**Pull request merged**